misc/chrome_extension/content-script.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+var port = chrome.runtime.connect();
+//global since it needs to persist beyond the function scope
+var elementsToHide = [];
+
+//Parses the XPath
```

Comment:
It's better to use jsdoc syntax to describe functions: 

```
/**
 * Parses the XPath
 * sample return
 * {type:"title", value:[element, element]}
 *
 * @param {string} title - the name of the metadata field
 * @param {string} xpath - the xpath to use to find the element in the page
 * @param {boolean} [getTrue] - set to true if you want the text value of the element found with xpath. (optional)
 * @returns {object}
 */
```

For more info on the syntax: http://usejsdoc.org/